tests/jvm/src/test/resources/manifest.metacp

```diff
@@ -12,12 +12,16 @@ Symbols:
 _empty_. => package _empty_
 _empty_.Test. => final object Test extends AnyRef { +1 decls }
   AnyRef => scala.AnyRef#
-_empty_.Test.main(). => method main(args: Array[String]): Unit
+_empty_.Test.main(). => method main(args: Array[scala.Predef.String]): Unit
```

Comment:
This change is kinda unfortunate, but it's also a consequence of how Scalac models objects. When object members are viewed from inside the object, they get the `Object.this` prefix. When they are viewed from outside the object, the get the `Object.` prefix. 

```
scala> object M { type T; def m: T = ??? }
defined object M

scala> object N { def n: M.T = ??? }
defined object N

scala> val m = symbolOf[M.type].info.decl(TermName("m")).asMethod
m: $r.intp.global.MethodSymbol = method m

scala> val n = symbolOf[N.type].info.decl(TermName("n")).asMethod
n: $r.intp.global.MethodSymbol = method n

scala> showRaw(m.returnType)
res17: String = TypeRef(ThisType(M), TypeName("T"), List())

scala> showRaw(n.returnType)
res18: String = TypeRef(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(SingleType(ThisType(<root>), $line43), $line43.$read), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), $iw), M), TypeName("T"), List())
```

I've no idea whether there is a distinction between the former and the latter, and in which contexts, so I propose we begin on the safe side: 1) classify ThisType as a trivial prefix, 2) classify its SingleType counterpart a non-trivial prefix.

Reply:
Alternatively, we could check equality, and if it returns `true` then classify both as trivial:

```
scala> m.returnType =:= n.returnType
res19: Boolean = true
```